### ISSUE TITLE
[modern tree-sitter] python attribute & method has changed scopes

### ISSUE THREAD

**Issue by asiloisad** [pulsar-edit/pulsar#576]
### Thanks in advance for your bug report!

- [X] Have you reproduced issue in [safe mode](https://pulsar-edit.dev/docs/atom-archive/faq/sections/what-does-safe-mode-do.html)?
- [X] Have you used the [debugging guide](https://pulsar-edit.dev/docs/atom-archive/hacking-atom/#debugging) to try to resolve the issue?
- [X] Have you checked our [FAQs](https://pulsar-edit.dev/docs/launch-manual/sections/faq/) to make sure your question isn't answered there?
- [X] Have you checked to make sure your issue does not already [exist](https://github.com/issues?q=is%3Aissue+user%3Apulsar-edit+is%3Aopen)?
- [X] Have you checked you are on the latest release of Pulsar?

### What happened?

The scopes of has changed and It look like one-light theme does not support it yet.

### Pulsar version

1.105.2023052201

### Which OS does this happen on?

🪟 Windows

### OS details

10 22H2

### Which CPU architecture are you running this on?

64-bit(x86_64)

### What steps are needed to reproduce this?

1. Open file with grammar Python
2. Type code

### Example 1:

`variable.other.object.property` -> `support.other.property.python`
`entity.name.function` -> `support.other.function.python`

1. old tree-sitter
![image](https://github.com/pulsar-edit/pulsar/assets/29706474/42d17f06-a95a-443b-b63f-2430e03bf46b)

2. modern tree-sitter
![image](https://github.com/pulsar-edit/pulsar/assets/29706474/65a1cfbd-d70f-4595-bad9-6f097aefa856)

### Example 2:

`entity.name.function.call` -> `support.other.function.python`

1. old tree-sitter
![image](https://github.com/pulsar-edit/pulsar/assets/29706474/85e33804-ad9e-48ce-b697-c8504da519fb)

2. modern tree-sitter
![image](https://github.com/pulsar-edit/pulsar/assets/29706474/09f070e1-101d-4f1b-a4d2-dd9f18dda72c)

**Comment by icecream17** [comment]
The same thing happens with javascript attributes

**Comment by asiloisad** [comment]
new scopes is a problem of compatibility with all availables themes, but if it is must-to-go-forward, then let's do it, but minimum requirements is to adapt base syntax themes (like one-dark & one-light). I'm not familiar with new scopes yet, but it look like the themes was not adopted to them yet.

**Comment by savetheclocktower** [comment]
Yeah, don't worry. Sometimes the answer is to change the scope back to what it was, and sometimes the answer is to change the theme, but we'll accommodate it either way. Rarely, the answer _might_ be to take no action, but this isn't one of those times.

To use the function example: most syntax themes just style `.function` directly and thus aren't affected by the renaming of `entity.name.function` to `support.other.function`, but this theme is an unintentional outlier, so I'll be sure to fix it.

**Comment by savetheclocktower** [comment]
Easy enough to change the color of the function invocation from blue to cyan-ish.

The stuff about properties is a bit harder. One of my design goals was to make it so that not every single identifier gets treated as a variable. In this case, I think that it leads to over-highlighting. When you write the JavaScript…

```javascript
foo.bar.baz = 'thud';
```

…in your editor and `foo`, `bar`, and `baz` are all red, I'm not sure what value that has. And if you were to write…

```javascript
foo.bar.troz();
```

…both `foo` and `bar` would be treated as variables, even if `foo` is something that you imported from another module.

My goal in JavaScript in particular (probably not as well realized in other grammars, but still an intended goal) was to scope things as `variable` only when they're declared, assigned, or reassigned. So in the following examples…

```
let foo;
foo = "troz";
bar.baz.foo = 19;
```

…only the `foo` tokens would be scoped as `variable`.

Now, since this is an opinionated choice, I'm fine with changing the syntax themes so that there are no visual regressions. The point is not to recolor everything and force people to deal with it; it's to make it _possible_ to scope things with more granularity than was previously possible. In the new grammar, `bar` and `baz` are scoped with `support.other.object` and `support.other.property`, hence they can be targeted and given the same styling as `variable` if that's what folks want. I'll leave this open for people's to weigh in, but my tentative plan is to do exactly this wherever it's needed in built-in syntax themes, and _perhaps_ to introduce a setting to control it _if_ I'm feeling ambitious.
